src/test/java/uk/gov/pay/directdebit/infra/PostgresTemplate.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.pay.directdebit.infra;
+
+import io.dropwizard.db.DataSourceFactory;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import static java.sql.DriverManager.getConnection;
+
+public class PostgresTemplate {
+
+    private static final String TEMPLATE_NAME = "templateddc";
+
+    public static void createTemplate(DataSourceFactory source) {
+        PostgresConfig config = PostgresConfig.valueOf(source);
+        try (Connection connection = getConnection(config.getDatabaseRootUri(), config.getUserName(), config.getPassword())) {
+            terminateDbConnections(connection, config.getDatabaseName());
+            connection.createStatement().execute("CREATE DATABASE " + TEMPLATE_NAME + " WITH TEMPLATE " + config.getDatabaseName() + " OWNER " + config.getUserName());
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    static void restoreTemplate(DataSourceFactory source) {
+        PostgresConfig config = PostgresConfig.valueOf(source);
+        try (Connection connection = getConnection(config.getDatabaseRootUri(), config.getUserName(), config.getPassword())) {
+            terminateDbConnections(connection, config.getDatabaseName());
+            connection.createStatement().execute("DROP DATABASE " + config.getDatabaseName());
+            connection.createStatement().execute("CREATE DATABASE " + config.getDatabaseName() + " WITH TEMPLATE " + TEMPLATE_NAME + " OWNER " + config.getUserName());
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void terminateDbConnections(Connection connection, String databaseName) throws SQLException {
+        connection.createStatement().execute("SELECT pg_terminate_backend(pg_stat_activity.pid) FROM pg_stat_activity"
+                + " WHERE pg_stat_activity.datname = '" + databaseName + "' AND pid <> pg_backend_pid()");
+    }
+
+    private static class PostgresConfig {
+
+        private final String username;
+        private final String password;
+        private final String databaseName;
+        private final String databaseRootUri;
+
+        private PostgresConfig(String username, String password, String databaseName, String databaseRootUri) {
+            this.username = username;
+            this.password = password;
+            this.databaseName = databaseName;
+            this.databaseRootUri = databaseRootUri;
+        }
+
+        private static final PostgresConfig valueOf(DataSourceFactory datasource) {
```

Comment:
I would call this something like `from` rather than `valueOf`, which is a bit misleading. Not a deal breaker, can change in another PR.

Reply:
Will do